### ISSUE TITLE
How to get the inference result after using 'enqueueV3'?

### ISSUE THREAD

**Issue by feiyuhuahuo** [NVIDIA/TensorRT#3128]
## Description

I'm trying to deploy a semantic segmentation model with TensorRT. I first converted the ONNX model to an engine. Then use 'enqueueV3'  to do inference. But I don't know whether it run successfully and  I don't know how to get the final segmentation mask.
Part of my code:

```
    IExecutionContext *context = engine->createExecutionContext();
    assert(context != nullptr);

    int32_t output_data[all_ios[1].pixel_num];
    int32_t output_data2[all_ios[2].pixel_num];

    for (int i = 0; i < 10; i++)
    {
        auto start = chrono::system_clock::now();

        vector<int32_t> img_vector = mat2vector("test2.bmp", all_ios[0].dim.d[1], all_ios[0].dim.d[0]);
        assert(img_vector.size() == all_ios[0].pixel_num);

        int32_t *img_data = img_vector.data();

        auto end = chrono::system_clock::now();
        auto time_read_img = chrono::duration_cast<chrono::milliseconds>(end - start).count();
        cout << "read img time:  " << time_read_img << endl;

        start = chrono::system_clock::now();

        void *input_mem{nullptr};
        void *output_mem1{nullptr};
        void *output_mem2{nullptr};

        CHECK(cudaMalloc(&input_mem, all_ios[0].size));
        CHECK(cudaMalloc(&output_mem1, all_ios[1].size));
        CHECK(cudaMalloc(&output_mem2, all_ios[2].size));
        cout << all_ios[0].size << endl;
        cout << all_ios[1].size << endl;
        cout << all_ios[2].size << endl;

        cudaStream_t stream;
        CHECK(cudaStreamCreate(&stream));

        context->setTensorAddress(all_ios[0].name, img_data);
        context->setTensorAddress(all_ios[1].name, output_mem1);
        context->setTensorAddress(all_ios[2].name, output_mem2);
        bool status = context->enqueueV3(stream);
        cout << status << endl;
    }

```
The `status` shows `1`. Does this mean the engine inference successfully?
 And I tried to add  `CHECK(cudaMemcpyAsync(output_data, output_mem1, all_ios[1].size, cudaMemcpyDeviceToHost, stream));` at the last line.
But the `CHECK` function tells it failed.

## Environment

**TensorRT Version**:
8.6.1.6

**NVIDIA GPU**:
RTX3090
**NVIDIA Driver Version**:
525.125.06
**CUDA Version**:
11.6
**CUDNN Version**:
8.9.2

Operating System:
UBUNTU22.04
Python Version (if applicable):

Tensorflow Version (if applicable):

PyTorch Version (if applicable):

Baremetal or Container (if so, version):

## Relevant Files
The total project is here: https://drive.google.com/drive/folders/1N-cscZ0uReMm2x2MuLd9OSjqNNgUAKka?usp=sharing
Just compile the 'main.cpp' and run.
**Model link**:

## Steps To Reproduce
Just compile the 'main.cpp'  in the above link and run.

**Comment by RecLusIve-F** [comment]
I think you should call cudaStreamSynchronize(stream) after enqueueV3. Then you can access the output.

**Comment by zerollzeng** [comment]
You can refer to the source code of trtexec, try search "enqueueV3"

**Comment by ttyio** [comment]
closing since no activity for more than 3 weeks, pls reopen if you still have question, thanks!
